File: app/src/main/java/com/faendir/lightning_launcher/multitool/util/SetupDistributor.java
package com.faendir.lightning_launcher.multitool.util;

import com.faendir.lightning_launcher.multitool.R;
import com.faendir.lightning_launcher.multitool.proxy.JavaScript;
import com.faendir.lightning_launcher.multitool.proxy.Utils;

/**
 * @author lukas
 * @since 09.07.18
 */
public class SetupDistributor implements JavaScript.Setup {
    private final Utils utils;

    public SetupDistributor(Utils utils) {
        this.utils = utils;
    }

    @Override
    public void setup() {
        String className = utils.getSharedPref().getString(utils.getString(R.string.pref_setupClass), null);
        if (className != null) {
            utils.getSharedPref().edit().putString(utils.getString(R.string.pref_setupClass), null).apply();
            ((Setup) new LightningObjectFactory(utils).get(className)).setup();
        }
    }
}

File: app/src/main/java/com/faendir/lightning_launcher/multitool/util/SetupDistributor.kt
package com.faendir.lightning_launcher.multitool.util

import com.faendir.lightning_launcher.multitool.R
import com.faendir.lightning_launcher.multitool.proxy.JavaScript
import com.faendir.lightning_launcher.multitool.proxy.Utils

/**
 * @author lukas
 * @since 09.07.18
 */
class SetupDistributor(private val utils: Utils) : JavaScript.Setup {
    override fun setup() {
        val className = utils.sharedPref.getString(utils.getString(R.string.pref_setupClass), null)
        if (className != null) {
            utils.sharedPref.edit().putString(utils.getString(R.string.pref_setupClass), null).apply()
            (LightningObjectFactory(utils)[className] as JavaScript.Setup).setup()
        }
    }
}
